main.go

```diff
@@ -3,60 +3,114 @@ package main
+
+	envPrefix = ""
 )
 
+func init() {
+	// Remove `-v` short option from --version flag
+	cli.VersionFlag.(*cli.BoolFlag).Aliases = nil
+}
+
 func main() {
-	ctx := context.Background()
-	err := sync(ctx)
+	ctx, stop, app := newApp()
+	defer stop()
+	err := app.RunContext(ctx, os.Args)
+	// If required flags aren't set, it will return with error before we could set up logging
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "ERROR: %v\n", err)
+		_, _ = fmt.Fprintf(os.Stderr, "%v\n", err)
 		os.Exit(1)
 	}
-	os.Exit(0)
 }
 
-func sync(ctx context.Context) error {
-	exoscaleApiKey, exoscaleApiSecret, k8sConfigs := cfg()
+func newApp() (context.Context, context.CancelFunc, *cli.App) {
+	logInstance := &atomic.Value{}
+	logInstance.Store(logr.Discard())
+	app := &cli.App{
+		Name:     appName,
+		Usage:    appLongName,
+		Version:  fmt.Sprintf("%s, revision=%s, date=%s", version, commit, date),
+		Compiled: compilationDate(),
 
-	// Fetch bucket name to namespace/tenant information lookup table from the configured k8s clusters
-	namespaceInfoByBucket, err := generateNamespaceLookupMap(ctx, k8sConfigs)
-	if err != nil {
-		return err
-	}
+		EnableBashCompletion: true,
 
-	// Fetch actual billing data from Exoscale
-	client, err := egoscale.NewClient(exoscaleApiKey, exoscaleApiSecret, egoscale.ClientOptWithAPIEndpoint("https://api-ch-gva-2.exoscale.com"))
-	if err != nil {
-		return err
+		Before: setupLogging,
+		Flags: []cli.Flag{
+			&cli.IntFlag{
+				Name: "log-level", Aliases: []string{"v"}, EnvVars: envVars("LOG_LEVEL"),
+				Usage: "number of the log level verbosity",
+				Value: 0,
+			},
+			&cli.StringFlag{
+				Name: "log-format", EnvVars: envVars("LOG_FORMAT"),
+				Usage:       "sets the log format (values: [json, console])",
+				DefaultText: "console",
+			},
+		},
+		Commands: []*cli.Command{
+			NewCommand(),
+		},
+		ExitErrHandler: func(ctx *cli.Context, err error) {
+			if err != nil {
+				AppLogger(ctx).Error(err, "fatal error")
+				cli.HandleExitCoder(cli.Exit("", 1))
+			}
+		},
 	}
-	resp, err := client.ListSosBucketsUsageWithResponse(ctx)
-	if err != nil {
-		return err
+	hasSubcommands := len(app.Commands) > 0
+	app.Action = rootAction(hasSubcommands)
+
+	parentCtx := context.WithValue(context.Background(), loggerContextKey{}, logInstance)
+	ctx, stop := signal.NotifyContext(parentCtx, syscall.SIGINT, syscall.SIGTERM)
+	return ctx, stop, app
+}
+
+func rootAction(hasSubcommands bool) func(context *cli.Context) error {
+	return func(ctx *cli.Context) error {
+		if hasSubcommands {
+			return cli.ShowAppHelp(ctx)
+		}
+		return LogMetadata(ctx)
 	}
-	for _, v := range *resp.JSON200.SosBucketsUsage {
-		fmt.Printf("name: %s, zoneName: %s, size: %d, createdAt: %s, namespace info: %s\n", *v.Name, *v.ZoneName, *v.Size, *v.CreatedAt, namespaceInfoByBucket[*v.Name])
+}
+
+// env combines envPrefix with given suffix delimited by underscore.
+func env(suffix string) string {
+	return envPrefix + "_" + suffix
```

Comment:
This is a bug if the prefix is empty (see also https://github.com/vshn/go-bootstrap/pull/51)
```suggestion
	return envPrefix + suffix
```